hyperlink/_url.py

```diff
@@ -160,6 +152,19 @@ def __nonzero__(self):
 _QUERY_DELIMS = _ALL_DELIMS - _QUERY_SAFE
 
 
+def _make_decode_map(delims, allow_percent=False):
+    ret = dict(_HEX_CHAR_MAP)
+    if not allow_percent:
+        delims = set(delims) | set([u'%'])
+    for delim in delims:
+        _hexord = hex(ord(delim))[2:].zfill(2).encode('ascii').upper()
```

Comment:
minor: I think you could simplify this by using the `%x` format tokens:

```python
_hexord = format(ord(delim), '02X').encode('ascii')
```

(I don’t know if there’s a performance difference, or if this code is particularly perf-critical)